path_oidc.go

```diff
@@ -134,21 +195,33 @@ func (b *jwtAuthBackend) pathCallback(ctx context.Context, req *logical.Request,
 		Scopes:       []string{oidc.ScopeOpenID},
 	}
 
+	var rawToken string
+	var oauth2Token *oauth2.Token
+
 	code := d.Get("code").(string)
-	if code == "" {
-		return logical.ErrorResponse(errLoginFailed + " OAuth code parameter not provided"), nil
+	if code == noCode {
```

Comment:
when would this occur? Trying to understand where the 'code' field would be submitted to this callback where it equals `no_code`

Reply:
When the UI POSTs to the plugin, the response HTML with the JS is configured here, which references `noCode`: https://github.com/hashicorp/vault-plugin-auth-jwt/pull/98/files#diff-21d8bdb7e1b596220622b0be6c0b8106R138 and pops it in the JS here: https://github.com/hashicorp/vault-plugin-auth-jwt/pull/98/files#diff-0e83cde1b3d66eaa8aa4e534959abebbR457.  So it's a sentinel value meaning go look in the cache state for 'code', if there is any. We can't leave that field out or even empty in the JS since the UI then complains.

Reply:
gotcha!